src/ConfigPostProcessor.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * @see       https://github.com/laminas/laminas-zendframework-bridge for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-zendframework-bridge/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-zendframework-bridge/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Laminas\ZendFrameworkBridge;
+
+class ConfigPostProcessor
+{
+    /** @var Replacements */
+    private $replacements;
+
+    public function __construct()
+    {
+        $this->replacements = new Replacements([
+            // Since we're rewriting at the PHP level, and not the raw
+            // strings from reading file contents, we need to add this
+            // one.
+            'zend-expressive' => 'expressive',
+        ]);
+    }
+
+    /**
+     * @return array
+     */
+    public function __invoke(array $config)
+    {
+        $rewritten = [];
+
+        foreach ($config as $key => $value) {
+            $newKey = is_string($key) ? $this->replacements->replace($key) : $key;
+
+            if (isset($rewritten[$newKey]) && is_array($rewritten[$newKey])) {
+                $rewritten[$newKey] = self::merge($rewritten[$newKey], $this->rewriteValue($value, $newKey));
```

Comment:
as discussed on slack - we need ensure that `$value` is array, not sure what we should do if one value is array and the other is not?

Reply:
Four options:

- Raise an exception
- Emit a notice
- Silently discard the old value (i.e., newer values take precedence over older ones)
- Use the following rules:
  - non-null values overwrite null values
  - array or object values overwrite scalar values
  - object values overwrite array values

And/or emit a notice AND do either the third or fourth option.

Not sure what users would expect here.

Reply:
I think we need to check what we are doing now on merging configuration and take the same approach here. Currently we can have the situation that in zf library we have:
```php
'zend-expressive' => [ ... array configuration ... ]
```

and user configuration:
```php
'zend-expressive' => 'string',
```

Reply:
I'll check and see what zend-config-aggregator does, and do that. :smile:

Reply:
zend-config-aggregator uses the same rules as zend-stdlib `ArrayUtils::merge`, which has the following algorithm:

- If both values are arrays, they're merged.
- Otherwise, the new value overwrites the old value.

This works fine for the config aggregator, because it processes configuration in a determined order (config providers, then global config files, then local config files).

For the post processor, it could potentially be problematic, as it depends on the internal array order: if we rewrite something with a ZF key and already have configuration under the equivalent Laminas key, it could potentially mean overwriting something that was not expected to be overwritten.

That said: in most cases, these will be coming from third-party components, which means that the keys should occur _earlier_ in the array.

I'm going to update to follow this algorithm, and put a note in the README about the scenario.

Reply:
We already follow that algorithm! But I'm adding a test case as well, to ensure we continue to do so.